libs/GameKit/source/games/GameSpeed.cpp

```diff
@@ -0,0 +1,112 @@
+
+
+#include "games/GameSpeed.h"
+
+#include "drivers/HighResClock.h"
+#include "rtos/ThisThread.h"
+
+using namespace leka;
+using namespace std::chrono;
+namespace {
+void do_nothing(leka::Position position) {}
+}	// namespace
+
+void GameSpeed::start()
+{
+	choseRandomCaptorAsTarget();
+
+	_startTime				 = rtos::Kernel::Clock::now();
+	auto update_touched_func = [&](Position position) { updateTouchedColor(position); };
+	_touch_sensor_kit.init();
+	_touch_sensor_kit.registerOnSensorTouched(update_touched_func);
+
+	_touch_sensor_kit.start();
+	_running = true;
+}
+
+void GameSpeed::run()
+{
+	if (_score == _scoreToWinLevel) {
+		_touch_sensor_kit.registerOnSensorTouched(do_nothing);
+		_ledManager.playReinforcer();
+
+		_running = false;
+
+	} else if (_win || !checkForTimeout()) {
```

Comment:
```suggestion
	} else if (target_touched || time_is_over) {
```

On aimerait pouvoir comprendre au premier coup d'oeil la condition. Je changerais carrément le nom de la fonction checkForTimeout() en isTimeout() car en donnant un bool, la fonction va répondre à ta question et tu pourras ainsi directement l'utiliser dans ta condition et que ça soit parlant

```suggestion
	} else if (target_touched || isTimeout()) {
```